File: CoreAPI/ViewModels/EmailVM.cs
using Core.Extensions;

namespace Core.ViewModels
{
    public class EmailVM
    {
        public string ConnKey { get; set; } = Utils.ConnKey;
        public string FromAddress { get; set; }
        public List<string> ToAddresses { get; set; }
        public List<string> CC { get; set; }
        public List<string> BCC { get; set; }
        public string Subject { get; set; }
        public string Body { get; set; }
        public List<string> PdfText { get; set; } = [];
        public List<string> FileName { get; set; } = [];
        public ICollection<string> Attachements { get; set; } = [];
        public ICollection<string> ServerAttachements { get; set; } = [];
    }
}

File: frontend/lib/models/emailVM.js
import { Client } from "../clients/client.js";

/**
 * Represents an email view model that encapsulates the email data.
 */
export class EmailVM {
    /**
     * Initializes a new instance of the EmailVM class with default values.
     */
    constructor() {
        /** @type {string} Connection key, default value fetched from Client.MetaConn */
        this.connKey = Client.MetaConn;

        /** @type {string} Email from address */
        this.fromAddress = '';

        /** @type {Array<string>} List of email addresses to send the email to */
        this.toAddresses = [];

        /** @type {Array<string>} List of email addresses for CC */
        this.cc = [];

        /** @type {Array<string>} List of email addresses for BCC */
        this.bcc = [];

        /** @type {string} Subject of the email */
        this.subject = '';

        /** @type {string} Body content of the email */
        this.body = '';

        /** @type {Array<string>} Texts that will be converted to PDF */
        this.pdfText = [];

        /** @type {Array<string>} Names of the files */
        this.fileName = [];

        /** @type {Set<string>} Collection of attachment identifiers */
        this.attachements = new Set();
    }
}